## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 447644`
Commit: [`5870ff2e850dcb10639647c861ab80b4c5d3edd5`](https://github.com/NixOS/nixpkgs/commit/5870ff2e850dcb10639647c861ab80b4c5d3edd5) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/5870ff2e850dcb10639647c861ab80b4c5d3edd5..pull/447644/head))
Merge: [`8e6f976f5c7861a51111cfde61893c2f8bde30ba`](https://github.com/NixOS/nixpkgs/commit/8e6f976f5c7861a51111cfde61893c2f8bde30ba)

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/18178262395


---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.firedrake</li>
    <li>python312Packages.firedrake.dist</li>
    <li>python313Packages.firedrake</li>
    <li>python313Packages.firedrake.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>haskellPackages.symengine-hs</li>
    <li>haskellPackages.symengine-hs.doc</li>
    <li>python312Packages.firedrake-fiat</li>
    <li>python312Packages.firedrake-fiat.dist</li>
    <li>python312Packages.symengine</li>
    <li>python312Packages.symengine.dist</li>
    <li>python313Packages.firedrake-fiat</li>
    <li>python313Packages.firedrake-fiat.dist</li>
    <li>python313Packages.symengine</li>
    <li>python313Packages.symengine.dist</li>
    <li>symengine</li>
    <li>symengine.dev</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.firedrake</li>
    <li>python312Packages.firedrake.dist</li>
    <li>python313Packages.firedrake</li>
    <li>python313Packages.firedrake.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>haskellPackages.symengine-hs</li>
    <li>haskellPackages.symengine-hs.doc</li>
    <li>python312Packages.firedrake-fiat</li>
    <li>python312Packages.firedrake-fiat.dist</li>
    <li>python312Packages.symengine</li>
    <li>python312Packages.symengine.dist</li>
    <li>python313Packages.firedrake-fiat</li>
    <li>python313Packages.firedrake-fiat.dist</li>
    <li>python313Packages.symengine</li>
    <li>python313Packages.symengine.dist</li>
    <li>symengine</li>
    <li>symengine.dev</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.firedrake-fiat</li>
    <li>python312Packages.firedrake-fiat.dist</li>
    <li>python312Packages.symengine</li>
    <li>python312Packages.symengine.dist</li>
    <li>python313Packages.firedrake-fiat</li>
    <li>python313Packages.firedrake-fiat.dist</li>
    <li>python313Packages.symengine</li>
    <li>python313Packages.symengine.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>haskellPackages.symengine-hs</li>
    <li>haskellPackages.symengine-hs.doc</li>
    <li>symengine</li>
    <li>symengine.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.symengine</summary>
<pre>E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../../store/yh2k6aryay5pmacnk98044ikihkbas6m-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:475
  /nix/store/yh2k6aryay5pmacnk98044ikihkbas6m-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-t60rqmm1&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../../store/yh2k6aryay5pmacnk98044ikihkbas6m-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:429
  /nix/store/yh2k6aryay5pmacnk98044ikihkbas6m-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-snaygz_7&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.25s =========================
ERROR: found no collectors for /nix/store/01479sw29p3kq8zf6r666hfayld3bnjv-python3.12-symengine-0.14.1/lib/python3.12/site-packages/symengine

Traceback (most recent call last):
  File &quot;/nix/build/nix-13855-837422580/source/empty/../bin/test_python.py&quot;, line 6, in &lt;module&gt;
    raise Exception(&#x27;Tests failed&#x27;)
Exception: Tests failed</pre>
</details>
<details>
<summary>python313Packages.symengine</summary>
<pre>E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../../store/blmjz22ir36m4c80nvdcg1k9c4mh11wa-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/blmjz22ir36m4c80nvdcg1k9c4mh11wa-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-8giycqxx&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../../store/blmjz22ir36m4c80nvdcg1k9c4mh11wa-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:429
  /nix/store/blmjz22ir36m4c80nvdcg1k9c4mh11wa-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-n_z8d23f&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.22s =========================
ERROR: found no collectors for /nix/store/av6wx9ph7ss8789qklbpm3ns3ljd8frz-python3.13-symengine-0.14.1/lib/python3.13/site-packages/symengine

Traceback (most recent call last):
  File &quot;/nix/build/nix-13855-837422581/source/empty/../bin/test_python.py&quot;, line 6, in &lt;module&gt;
    raise Exception(&#x27;Tests failed&#x27;)
Exception: Tests failed</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>python312Packages.firedrake</li>
    <li>python312Packages.firedrake-fiat</li>
    <li>python312Packages.firedrake-fiat.dist</li>
    <li>python312Packages.firedrake.dist</li>
    <li>python312Packages.symengine</li>
    <li>python312Packages.symengine.dist</li>
    <li>python313Packages.firedrake</li>
    <li>python313Packages.firedrake-fiat</li>
    <li>python313Packages.firedrake-fiat.dist</li>
    <li>python313Packages.firedrake.dist</li>
    <li>python313Packages.symengine</li>
    <li>python313Packages.symengine.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>haskellPackages.symengine-hs</li>
    <li>haskellPackages.symengine-hs.doc</li>
    <li>symengine</li>
    <li>symengine.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.symengine</summary>
<pre>E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../../store/3blbsz1zw5llvsgzk7n9ibjwyd4dqqwq-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:475
  /nix/store/3blbsz1zw5llvsgzk7n9ibjwyd4dqqwq-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-zjupyj62&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../../store/3blbsz1zw5llvsgzk7n9ibjwyd4dqqwq-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:429
  /nix/store/3blbsz1zw5llvsgzk7n9ibjwyd4dqqwq-python3.12-pytest-8.4.1/lib/python3.12/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-bx0dm0wo&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.10s =========================
ERROR: found no collectors for /nix/store/1kiaym2ba3rri6g6iv1g3331h4misziw-python3.12-symengine-0.14.1/lib/python3.12/site-packages/symengine

Traceback (most recent call last):
  File &quot;/nix/build/nix-5718-3623427250/source/empty/../bin/test_python.py&quot;, line 6, in &lt;module&gt;
    raise Exception(&#x27;Tests failed&#x27;)
Exception: Tests failed</pre>
</details>
<details>
<summary>python313Packages.symengine</summary>
<pre>E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../../store/wi545s8n8fr6wylzs5r5asa168s1ndqq-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/wi545s8n8fr6wylzs5r5asa168s1ndqq-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-e3swschp&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../../store/wi545s8n8fr6wylzs5r5asa168s1ndqq-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:429
  /nix/store/wi545s8n8fr6wylzs5r5asa168s1ndqq-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-48lgb31x&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.08s =========================
ERROR: found no collectors for /nix/store/acb7wfpacbsr1mnmgm6k0grbma0p89xw-python3.13-symengine-0.14.1/lib/python3.13/site-packages/symengine

Traceback (most recent call last):
  File &quot;/nix/build/nix-5718-3623427251/source/empty/../bin/test_python.py&quot;, line 6, in &lt;module&gt;
    raise Exception(&#x27;Tests failed&#x27;)
Exception: Tests failed</pre>
</details>
</details>

--- a/pkgs/by-name/sy/symengine/package.nix
+++ b/pkgs/by-name/sy/symengine/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   cmake,
   gmp,
   flint,
@@ -24,6 +25,18 @@ stdenv.mkDerivation (finalAttrs: {
   outputs = [
     "out"
     "dev"
+  ];
+
+  # upgrade supported cmake version in SymEngineConfig.cmake
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/symengine/symengine/commit/c149b874b8ff947e51e8e58670a0d37daf588f86.patch?full_index=1";
+      hash = "sha256-LqkJRPdsbE8OE8G6AkpWX9B+GqnOQjUNPHpKKIcCL3Q=";
+    })
+    (fetchpatch2 {
+      url = "https://github.com/symengine/symengine/commit/186f72e208220efd12362c336a49378076f63f30.patch?full_index=1";
+      hash = "sha256-CuQra9K3MTxm8M0bt3LooJz9HgW0/Jy6ydRBCvEgkO4=";
+    })
   ];
 
   nativeBuildInputs = [ cmake ];

--- a/pkgs/by-name/sy/symengine/package.nix
+++ b/pkgs/by-name/sy/symengine/package.nix
@@ -7,17 +7,17 @@
   flint,
   mpfr,
   libmpc,
-  withShared ? true,
+  withShared ? !stdenv.hostPlatform.isStatic,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "symengine";
   version = "0.14.0";
 
   src = fetchFromGitHub {
     owner = "symengine";
     repo = "symengine";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-WriVcYt3fkObR2U4J6a4KGGc2HgyyFyFpdrwxBD+AHA=";
   };
 
@@ -31,28 +31,23 @@ stdenv.mkDerivation rec {
   ];
 
   cmakeFlags = [
-    "-DWITH_FLINT=ON"
-    "-DINTEGER_CLASS=flint"
-    "-DWITH_SYMENGINE_THREAD_SAFE=yes"
-    "-DWITH_MPC=yes"
-    "-DBUILD_FOR_DISTRIBUTION=yes"
-  ]
-  ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [
-    # error: unrecognized instruction mnemonic, did you mean: bit, cnt, hint, ins, not?
-    "-DBUILD_TESTS=OFF"
-  ]
-  ++ lib.optionals withShared [
-    "-DBUILD_SHARED_LIBS=ON"
+    (lib.cmakeBool "WITH_FLINT" true)
+    (lib.cmakeFeature "INTEGER_CLASS" "flint")
+    (lib.cmakeBool "WITH_SYMENGINE_THREAD_SAFE" true)
+    (lib.cmakeBool "WITH_MPC" true)
+    (lib.cmakeBool "BUILD_FOR_DISTRIBUTION" true)
+    (lib.cmakeBool "BUILD_TESTS" finalAttrs.finalPackage.doCheck)
+    (lib.cmakeBool "BUILD_SHARED_LIBS" withShared)
   ];
 
-  doCheck = true;
+  # error: unrecognized instruction mnemonic, did you mean: bit, cnt, hint, ins, not?
+  doCheck = !(stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64);
 
-  meta = with lib; {
+  meta = {
     description = "Fast symbolic manipulation library";
     homepage = "https://github.com/symengine/symengine";
-    platforms = platforms.unix ++ platforms.windows;
-    license = licenses.bsd3;
-    maintainers = [ maintainers.costrouc ];
+    platforms = with lib.platforms; unix ++ windows;
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ costrouc ];
   };
-
-}
+})

--- a/pkgs/development/python-modules/symengine/default.nix
+++ b/pkgs/development/python-modules/symengine/default.nix
@@ -3,19 +3,17 @@
   buildPythonPackage,
   fetchFromGitHub,
   cython,
+  setuptools,
   cmake,
   symengine,
   pytest,
   sympy,
   python,
-  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "symengine";
   version = "0.14.1";
-
-  build-system = [ setuptools ];
   pyproject = true;
 
   src = fetchFromGitHub {
@@ -25,19 +23,24 @@ buildPythonPackage rec {
     hash = "sha256-adzODm7gAqwAf7qzfRQ1AG8mC3auiXM4OsV/0h+ZmUg=";
   };
 
-  env = {
-    SymEngine_DIR = "${symengine}";
-  };
-
   postPatch = ''
     substituteInPlace setup.py \
       --replace-fail "'cython>=0.29.24'" "'cython'"
   '';
 
+  build-system = [
+    cython
+    setuptools
+  ];
+
   dontUseCmakeConfigure = true;
+
   nativeBuildInputs = [
     cmake
-    cython
+  ];
+
+  buildInputs = [
+    symengine
   ];
 
   nativeCheckInputs = [
@@ -47,15 +50,17 @@ buildPythonPackage rec {
 
   checkPhase = ''
     runHook preCheck
+
     mkdir empty && cd empty
     ${python.interpreter} ../bin/test_python.py
+
     runHook postCheck
   '';
 
-  meta = with lib; {
+  meta = {
     description = "Python library providing wrappers to SymEngine";
     homepage = "https://github.com/symengine/symengine.py";
-    license = licenses.mit;
+    license = lib.licenses.mit;
     maintainers = [ ];
   };
 }

